### PR TITLE
added option to use yarn or npm

### LINKED ISSUE #30: yarn option
Add the option to scaffold the project with yarn or npm

### PR DESCRIPTION
**Pull request by CrimsonNynja**
resolves #30

### CODE REVIEW

**Review by dozyhermit — APPROVED** [review]
this is fine, but i wonder if maybe it's easier to just use yarn as a default? 

i believe yarn respects the npm lock file, but npm does not respect the yarn lock file. (or maybe they respect each-other now? i can't remember.)

File: generators/app/index.js [hunk 1]
```diff
@@ -61,6 +61,12 @@ module.exports = class extends Generator {
         default: 'hwWxD5cB6LtaCB0GOcbaxiOI2eaFoC4rIT9jh51DCdB6p9IZrHTMRuFUM72xIjm',
         when: (answers) => answers.auth === 'JWT',
       },
+      {
+        type: 'list',
+        name: 'packageManager',
+        message: 'which package manager do you wish to use?',
```
Comment (dozyhermit) [inline generators/app/index.js]: semantic, but "which package manager do you want to use?" over "which package manager do you wish to use?"

File: generators/app/index.js [hunk 2]
```diff
@@ -71,9 +77,19 @@ module.exports = class extends Generator {
     this.log('DbName', this.questions.DbName);
     this.log('auth', this.questions.auth);
     this.log('secretKey', this.questions.secretKey);
+    this.log('packageManager', this.questions.packageManager);
   }
 
   writing() {
+    const install = (packages, options) => {
+      if (this.options.packageManager === 'npm') {
+        this.npmInstall(packages, options);
```
Comment (dozyhermit) [inline generators/app/index.js:86]: return after this; exit early and fast.

File: generators/app/index.js [hunk 3]
```diff
@@ -71,9 +77,19 @@ module.exports = class extends Generator {
...
+      }
+      if (this.options.packageManager === 'yarn') {
+        this.yarnInstall(packages, options);
```
Comment (dozyhermit) [inline generators/app/index.js:89]: same as above here

File: generators/app/index.js [hunk 4]
```diff
@@ -211,6 +227,20 @@ module.exports = class extends Generator {
   }
 
   install() {
-    this.installDependencies({ bower: false });
+    if (this.options.packageManager === 'npm') {
```
Comment (dozyhermit) [inline generators/app/index.js:235]: same deal as above; just return here.

File: generators/app/index.js (hunk 2 above)
Comment (CrimsonNynja) [reply generators/app/index.js:86]: Cool catch, origionally had bower as a 3rd option, but removed it

**Comment by dozyhermit** [comment]
oooh, you could also do something a little cooler 

```
const packageManager = (this: any) = { 
  yarn: this.yarnInstall,
  npm: this.npmInstall
};


const install = (packages, options) => {
    const man = this.options.packageManager;
    const install = packageManager(this);
    install[man](packages, options);
}
```

or something. just a little quicker than adding new ifs each time. (you might not be able to pass `this` through to the function, but you get the idea.)

**Pull request merged**